### ISSUE TITLE
Glue Job Deployer: Glue Connection AZ error

### ISSUE THREAD

**Issue by cnfait** [awslabs/aws-serverless-data-lake-framework#317]
**Describe the bug**
Initially reported in https://github.com/awslabs/aws-serverless-data-lake-framework/issues/312#issuecomment-2049059921

```
JobName:sdlf-team-legislators and JobRunId:jr_4ea9c23e577346ee8530c3ba84210b6fd9fd50e7c9 failed to execute with exception Availability Zone us-east-1a does not correspond to subnet (Service: AWSGlueJobExecutor; Status Code: 400; Error Code: InvalidInputException; Request ID: 1e56b9d1-6f0c-404a-b9e8-3530c2f9b977; Proxy: **null)**
```

Due to
```
        PhysicalConnectionRequirements:
          AvailabilityZone: us-east-1a
```

**SDLF release (if known):**
2.2.0

**Additional context**
Perhaps `AvailabilityZone: !Sub ${AWS::Region}a` can help.

**Comment by dverdonschot** [comment]
So when rolling out the GlueJob deployer we had it fail on rGlueConnection like this:
Template error: Fn::Select cannot select nonexistent value at index 1

In the end we changed the SubnetId Select value from 1 to 0:

```yaml
  rGlueConnection:
          SubnetId: !Select
            - "0"
            - !Split [",", !ImportValue sdlf-cicd-domain-roles-vpc-subnets]
```

**Comment by cnfait** [comment]
The diff in GitHub is a bit messy due to moving stuff around but this should be fixed in #319.

To update a current installation:
* make sure you have an up to date clone of the SDLF repository (in particular `sdlf-cicd`)
* rerun the `./deploy.sh devops-account ...` command
* update your CodeCommit `sdlf-cicd` repository with the content of the upstream `sdlf-cicd` (for all relevant branches, such as `dev`), wait for the relevant `rMain`CodePipeline to complete

Then the deployment of Glue Jobs should work and the correct AZ should be automatically specified in the template.

**Comment by cnfait** [comment]
One more PR coming up for this issue.
